### PR TITLE
[WL-1096]: transfer SUI token

### PR DESCRIPTION
**Pull request by FME849**

### CODE REVIEW

File: apps/wallet/src/engine/runners/sui/tokens.ts [hunk 1]
```diff
@@ -9,22 +13,50 @@ export const constructSuiTokenDocument = async (
...
+			if (!coinList[coinObject.coinType]) {
+				coinList[coinObject.coinType] = [coinObject];
+			} else {
+				coinList[coinObject.coinType].push(coinObject);
+			}
+
+			return coinList;
+		},
+		{} as Record<string, CoinStruct[]>,
+	);
```
Comment (tanlethanh) [inline apps/wallet/src/engine/runners/sui/tokens.ts:61]: you might want to use `Object.groupBy`

File: apps/wallet/src/features/Send/InputTransaction/TransactionFee/DefaultFee.tsx [hunk 2]
```diff
@@ -23,6 +26,12 @@ export const DefaultTransactionFee: FC<Props> = () => {
 
 	const feeString = `${feeAmount ? feeAmount : 0} ${networkToken}`;
 
+	useEffect(() => {
+		if (network && (token || nft) && receiver && Number(amount) > 0) {
+			txActions.updateTransactionFee();
```
Comment (tanlethanh) [inline apps/wallet/src/features/Send/InputTransaction/TransactionFee/DefaultFee.tsx:31]: I think it needed to wrap this function by throttle

File: apps/wallet/src/utils/transaction/construct.ts [hunk 3]
```diff
@@ -64,3 +66,39 @@ export const constructSolanaSendNftTransaction = async (
...
+				sourceCoins.map((coin) => coin.coinObjectId),
+			);
+		}
+		coin = txb.splitCoins(primaryCoin, [
+			txb.pure(amount * 10 ** token.decimals),
+		]);
+	}
+	txb.transferObjects([coin], txb.pure(receiver, 'address'));
+	txb.setSenderIfNotSet(sender);
+
```
Comment (tanlethanh) [inline apps/wallet/src/utils/transaction/construct.ts:102]: you could move this construct logic to `@walless/network`

File: apps/wallet/src/engine/runners/sui/tokens.ts (hunk 1 above)
Comment (FME849) [reply apps/wallet/src/engine/runners/sui/tokens.ts:61]: Wow! This's great, idk it exist

File: apps/wallet/src/features/Send/InputTransaction/TransactionFee/DefaultFee.tsx (hunk 2 above)
Comment (FME849) [reply apps/wallet/src/features/Send/InputTransaction/TransactionFee/DefaultFee.tsx:31]: I assume 300ms was enough. Additionally, I refactor the `txActions.updateTransactionFee` to exclude the transfer amount

**Review by tanlethanh — APPROVED** [review]

**Pull request merged**